xaitk_saliency/utils/detection.py

```diff
@@ -0,0 +1,56 @@
+from typing import Optional
+
+import numpy as np
+
+
+def format_detection(
+    bbox_mat: np.ndarray,
+    classification_mat: np.ndarray,
+    objectness: Optional[np.ndarray] = None,
+) -> np.ndarray:
+    """
+    Combine detection and classification output, with optional objectness
+    output, into the combined format required for
+    :py:meth:`.GenerateDetectorProposalSaliency.generate` ``*_dets`` input
+    parameters.
+
+    We enforce some shape consistency so that we can create a valid output
+    matrix. The input bounding box matrix should be of shape `[nDets x 4]`, the
+    classification matrix should be of shape `[nDets x nClasses]`, and the
+    objectness vector, if provided, should be of size `nDets`.
+
+    If an `objectness` score vector is not provided, we assume a vector of 1's.
+
+    The output of this function is a matrix that is of shape
+    `[nDets x (4+1+nClasses)]`. This is the result of horizontally stacking the
+    input in bbox, objectness and classification order. The output matrix
+    data-type will follow numpy's rules about safe-casting given the
+    combination of input matrix types.
+
+    In exceptions about shape mismatches, index 0 refers to the `bbox_mat`
+    input, index 1 refers to the ``objectness` vector, and index 2 refers to
+    the `classification_mat`.
+
+    :param bbox_mat: Matrix of bounding boxes. This matrix should have the
+        shape `[nDets x 4]`. The format of each row-vector is not important
+        but generally expected to be `[left, top, right, bottom]` pixel
+        coordinates. This matrix will must be of a type that is float-castable.
+    :param classification_mat: Matrix of classification results from the
```

Comment:
Not sure if this should be clarified here, but `classification_mat` is essentially the confidences/logits from the classification, so these will also be in the range [0,1].

```suggestion
    :param classification_mat: Matrix of classification scores from the
```

Reply:
fixed